editor/src/components/context-menu-wrapper.tsx

```diff
@@ -217,45 +218,45 @@ export class InspectorContextMenuWrapper<T> extends ReactComponent<ContextMenuWr
           },
         }}
       >
-        {this.props.items.length > 0 ? (
-          <React.Fragment>
-            <MenuProvider
-              key={`${this.props.id}-provider`}
-              id={this.props.id}
-              style={{
-                width: '100%',
-                height: '100%',
-              }}
-            >
-              {this.props.children}
-            </MenuProvider>
-            <MomentumContextMenu
-              key={`${this.props.id}`}
-              id={this.props.id}
-              items={this.props.items}
-              getData={this.getData}
-            />
-          </React.Fragment>
-        ) : (
-          this.props.children
-        )}
+        <React.Fragment>
+          <MenuProvider
+            key={`${this.props.id}-provider`}
+            id={this.props.id}
+            itemsLength={this.props.items.length}
+            style={{
+              width: '100%',
+              height: '100%',
+            }}
+          >
+            {this.props.children}
+          </MenuProvider>
+          <MomentumContextMenu
+            key={`${this.props.id}`}
+            id={this.props.id}
+            items={this.props.items}
+            getData={this.getData}
+          />
+        </React.Fragment>
       </div>
     )
   }
 }
 
 interface MenuProviderProps {
   id: string
+  itemsLength: number
   style?: React.CSSProperties
 }
 
 export const MenuProvider: React.FunctionComponent<MenuProviderProps> = (props) => {
```

Comment:
I think we could change this to use `PropsWithChildren` here to save having to explicitly pass in the `itemsLength`

Reply:
Oh no, I misunderstood, ignore that